File: JXBC-Server/BossComments/apiserver/www/controller/AppController.php
<?php
namespace app\www\controller;
use think\Config;
use think\Request;
use think\Controller;
use app\common\controllers\AuthenticatedController;

class AppController extends Controller
{
    public function index(Request $request)
    {
        $this->redirect("/");
    }

    public function download(Request $request)
    {
        $target = $request->get("target");
        $userAgent = strtolower($request->header('user-agent'));
        if($target == "wechat" || strpos($userAgent, "micromessenger") !== false) {
            $this->redirect(Config::get("app_download_wechat"));
        }else if($target == "ios" || strpos($userAgent, "ios") !== false  || strpos($userAgent, "iphone") !== false || strpos($userAgent, "mac os x") !== false) {
            $this->redirect(Config::get("app_download_ios"));
        } else if ($request->isMobile()&& !empty(Config::get("app_download_apk_mobile"))) {
            $this->redirect(Config::get("app_download_apk_mobile"));
        } else {
            $this->redirect(Config::get("app_download_apk"));
        }
    }
}

File: JXBC-Server/appbase/src/JXBC.Web/JXBC.WebSite/Controllers/AppController.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Web;
using System.Web.Mvc;
using JXBC.WebCore.ViewModels;
using JXBC.WebCore;
using M2SA.AppGenome;
using JXBC.Packages.ApkProviders;
using M2SA.AppGenome.Logging;
using JXBC.Packages;

namespace JXBC.WebSite.Controllers
{
    /// <summary>
    /// 
    /// </summary>
    public class AppController : Controller
    {
        /// <summary>
        /// 
        /// </summary>
        /// <returns></returns>
        [HttpGet]
        public ActionResult index(string name)
        {
            return View("error");
        }

        [HttpGet]
        public ActionResult Download(string target = null)
        {
            var userAgent = Request.UserAgent.ToLower();
            if (target == "ios" || userAgent.Contains("iphone") || userAgent.Contains("mac os x"))
            {
                var downloadUrl = AppEnvironment.GetValueFromConfig("appstore:ios", "/");
                return Redirect(downloadUrl);
            }
            else if(Request.UserAgent.ToLower().Contains("micromessenger") || target == "wechat")
            {
                var appVersion = GetLastVersionByAndroid();                
                var downloadUrl = string.Format("http://a.app.qq.com/o/simple.jsp?pkgname={0}", appVersion.PackageName);
                return Redirect(downloadUrl);
            }
            else
            {
                var file = PackageHelper.GetAndroidPackageFile();
                PackageHelper.DownloadFile(PackageHelper.GetAndroidPackageFile(), this.Response);
                return null;
            }
        }

        [HttpGet]
        public ActionResult LastVersion()
        {            
            var appVersion = GetLastVersionByAndroid();
            return Json(appVersion, JsonRequestBehavior.AllowGet);
        }

        private AppVersion GetLastVersionByAndroid()
        {
            var apkFile = PackageHelper.GetAndroidPackageFile();
            var appVersion = PackageManager.GetLastVersion(apkFile);
            appVersion.DownloadUrl = this.Request.Url.AbsoluteUri.ToLower().Replace("LastVersion".ToLower(), "Download");
            return appVersion;
        }
    }
}
